src/main/java/tool/designpatterns/verifiers/multiclassverifiers/decorator/DecoratorVerifier.java

```diff
@@ -0,0 +1,226 @@
+     *             <ul>
+     *             <li>It must house a component field</li>
+     *             <li>The component field needs to be initialized during
+     *             construction</li>
+     *             </ul>
+     *         </li>
+     *         <li>Every concrete decorator must extend an existing abstract
+     *         decorator in the
+     *         current the pattern instance</li>
+     *      </ol>
+     * </p>
+     *
+     * @param patternInstance An instance of the decorator pattern to be verified
+     *
+     * @return A {@link Feedback} object that contains the result and information regarding whether
+     *     ther or not the instance of the pattern was valid
+     */
+    public Feedback verify(DecoratorPatternInstance patternInstance) {
+        Feedback allElementsChild = patternInstance.hasAllElements(patternInstance);
+        List<Feedback> childFeedbacks = new ArrayList<>();
+        childFeedbacks.add(allElementsChild);
+        if (!allElementsChild.getIsError()) {
+            childFeedbacks.add(interfaceContainsMethod(patternInstance.interfaceComponent));
+            patternInstance.abstractDecorators.forEach(decorator -> {
+                childFeedbacks.add(hasFieldOfType(decorator, patternInstance.interfaceComponent));
+                childFeedbacks.add(
+                    fieldInitializedInConstr(decorator, patternInstance.interfaceComponent));
+            });
+            patternInstance.concreteDecorators.forEach(decorator -> {
+                childFeedbacks.add(
+                    fieldInitializedInConstr(decorator, patternInstance.interfaceComponent));
+            });
+        }
+
+        return Feedback.getPatternInstanceFeedback(childFeedbacks);
+    }
+
+    /**
+     * Checks that an interfaceComponent contains at least one public method.
+     *
+     * @param interfaceComponent The interface of the interfaceComponent
+     *
+     * @return A {@link Feedback} object containing the result
+     */
+    public Feedback interfaceContainsMethod(
+        ClassOrInterfaceDeclaration interfaceComponent) {
+        Feedback result;
+        MethodDeclaration erroringMethod = null;
+
+        AtomicBoolean resultBool = new AtomicBoolean(false);
+        for (MethodDeclaration methodDeclaration : interfaceComponent.findAll(
+            MethodDeclaration.class)) {
+            if (!methodDeclaration.isPrivate()) {
+                resultBool.set(true);
+                erroringMethod = methodDeclaration;
+            }
+        }
+
+        if (resultBool.get()) {
+            result = Feedback.getSuccessfulFeedback();
+        } else {
+            String msg = "Interface does not contain any methods!";
+            if (erroringMethod == null) {
+                result = Feedback.getPatternInstanceNoChildFeedback(msg);
+            } else {
+                result = Feedback.getNoChildFeedback(msg, new FeedbackTrace(erroringMethod));
+            }
```

Comment:
You never call this method with a MethodDeclaration as an argument and erroringMethod is therefore always null if the result is false. This method could either be simplified or you may have missed a step in some verification.

Reply:
During the change to feedback some misinterpretation of the code happened which resulted in this. Resolved the unaccessible block aswell as restructured the code that caused this confusion in the first place